work_queue/src/work_queue.c

```diff
@@ -3965,6 +3969,56 @@ static struct work_queue_worker *find_worker_by_time(struct work_queue *q, struc
 	}
 }
 
+
+// checks workers against tasks and returns if a task has the  
+// possibility of fitting a worker
+static int check_task_fit_in_worker(struct work_queue *q, struct work_queue_task *t, struct work_queue_worker *w)
```

Comment:
@Cpreciad  We want to compare against all the possible declared resources for the task. Things get a little bit more complicated when resources are declared in a category of tasks rather than directly with  `t->resources_requested`.

What this means is that you have to do (just after the `int ok = 1;`):

```C
    struct rmsummary *l = task_worker_box_size(q, w, t);
```

And then anywhere you use `t->resources_requested` use simply `l`.

Finally, just before the `return ok;`, you need to add:

```C
rmsummary_delete(l);
```

Reply:
made the change, and ensured that an unfit task was printed to console